### ISSUE TITLE
Same method invocation within a chain of invocation (fluent interface)

### ISSUE THREAD

**Issue by dgroup** [dgroup/arch4u-pmd#142]
### Summary
| Key            |  Value |
|:---------------|:------|
| Rule           |  AvoidSameMethodInvocationInChain  |
| arch4u version       | 0.3.0 |

### Your rule definition
```xml
<rule name="AvoidSameMethodInvocationInChain"
      since="0.3.0"
      language="java"
      externalInfoUrl="https://github.com/dgroup/arch4u-pmd/discussions/..."
      message="Avoid same method invocation in chain: https://github.com/dgroup/arch4u-pmd/discussions/..."
      class="io.github.dgroup.arch4u.pmd.AvoidSameMethodInvocationInChain">
    <priority>3</priority>
    <properties>
      <property name="allowedMethods" value="to"/>
    </properties>
</rule>
```

### Describe the defect or false/positive rule violation
- [ ] Case 1: method `description` invoked twice
```java
        rest()
            .post("/tbd")
            .id("tbd-endpoint").description("TBD endpoint")
            .description("TBD endpoint again ") //violation here 
            .consumes(MediaType.APPLICATION_JSON_VALUE)
            .produces(MediaType.APPLICATION_JSON_VALUE)
```
- [ ] Case 2: method `to` is allowed
```java
        rest()
            .post("/tbd")
            .id("tbd-endpoint")
            .description("tbd endpoint")
            .consumes(MediaType.APPLICATION_JSON_VALUE)
            .produces(MediaType.APPLICATION_JSON_VALUE)
            .to("tbd 1")
            .to("tbd 2")
```
